packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -532,35 +534,83 @@ export const compileExplore = ({
             );
         }
     });
-    const joinedTableNames = [baseTable, ...joinedTables.map((j) => j.table)];
-    const joined = joinedTableNames.reduce(
-        (prev, tableName) => ({ ...prev, [tableName]: tables[tableName] }),
-        {},
+    const aliases = [
+        baseTable,
+        ...joinedTables.map((join) => join.alias || join.table),
+    ];
+    if (aliases.length !== new Set(aliases).size) {
+        throw new CompileError(
+            `Failed to compile explore "${name}". Cannot join to the same table multiple times table in an explore. Use an 'alias'`,
+            {},
+        );
+    }
+    const includedTables = joinedTables.reduce<Record<string, Table>>(
+        (prev, join) => ({
+            ...prev,
+            [join.alias || join.table]: {
+                ...tables[join.table],
+                name: join.alias || tables[join.table].name,
+                label:
+                    join.label ||
+                    (join.alias && friendlyName(join.alias)) ||
+                    tables[join.table].label,
+                dimensions: Object.keys(tables[join.table].dimensions).reduce<
+                    Record<string, Dimension>
+                >(
+                    (prevDimensions, dimensionKey) => ({
+                        ...prevDimensions,
+                        [dimensionKey]: {
+                            ...tables[join.table].dimensions[dimensionKey],
+                            table: join.alias || tables[join.table].name,
+                            tableLabel:
+                                join.label ||
+                                (join.alias && friendlyName(join.alias)) ||
+                                tables[join.table].label,
```

Comment:
Suggestion: would be cleaner if we defined these variables at the top since we repeat this 3 or 4 times. 
We can wait for e2e tests to pass, make this change and merge if unit tests pass.